webhook/webhook.go

```diff
@@ -136,6 +136,43 @@ type GenericCRD interface {
 	runtime.Object
 }
 
+// NewAdmissionController constructs an AdmissionController
+func NewAdmissionController(
+	client kubernetes.Interface,
+	opts ControllerOptions,
+	handlers map[schema.GroupVersionKind]GenericCRD,
+	logger *zap.SugaredLogger,
+	ctx func(context.Context) context.Context,
+	disallowUnknownFields bool) *AdmissionController {
+
+	reporter, err := NewStatsReporter()
+	if err != nil {
+		panic(err)
+	}
+
+	return NewAdmissionControllerWithStats(client, opts, handlers, logger, &reporter, ctx, disallowUnknownFields)
+}
+
+// NewAdmissionControllerWithStats constructs an AdmissionController where the caller provides the StatsReporter
```

Comment:
Is this used anywhere outside this file?

Reply:
In the tests now and consuming packages can choose to use these instead of initializing the struct themselves.

I plan on having knative/serving use this constructor when I pull in the changes.

Reply:
Sounds good.